snykTask/src/index.ts

```diff
@@ -146,21 +153,39 @@ async function runSnykTest(
 
   const snykTestToolRunner = tl
     .tool(snykPath)
+    .argIf(taskArgs.testType == TestType.CODE, 'code')
+    .argIf(
+      taskArgs.dockerImageName || taskArgs.testType == TestType.CONTAINER_IMAGE,
+      'container',
+    )
     .arg('test')
     .argIf(
-      taskArgs.severityThreshold,
+      taskArgs.testType != TestType.CODE && taskArgs.severityThreshold,
       `--severity-threshold=${taskArgs.severityThreshold}`,
     )
-    .argIf(taskArgs.dockerImageName, `--docker`)
+    .argIf(
+      taskArgs.testType == TestType.CODE && taskArgs.codeSeverityThreshold,
+      `--severity-threshold=${taskArgs.codeSeverityThreshold}`,
+    )
     .argIf(taskArgs.dockerImageName, `${taskArgs.dockerImageName}`)
     .argIf(fileArg, `--file=${fileArg}`)
     .argIf(taskArgs.ignoreUnknownCA, `--insecure`)
     .argIf(taskArgs.organization, `--org=${taskArgs.organization}`)
     .argIf(taskArgs.projectName, `--project-name=${projectNameArg}`)
     .arg(`--json-file-output=${jsonReportOutputPath}`)
-    .argIf(isDebugMode(), '-d')
+    .argIf(isDebugMode() && taskArgs.testType != TestType.CODE, '-d')
+    .argIf(taskArgs.testType == TestType.CODE, `--json`)
     .line(taskArgs.additionalArguments);
 
+  // pipe to json file output with --json to handle no code issues found
+  if (taskArgs.testType == TestType.CODE) {
+    const catToolRunner = tl.tool('cat');
```

Comment:
question: Is this available on all possible runners & operating systems?

Reply:
valid point. this requires a switch to `echo` which is available to both windows and unix-based operating systems. The call of `pipeExecOutputToTool` is for writing console output stream to a json file for no-code-issue-found use-case (where the json is not auto-generated with `--json-file-output`, hence `--json` is also applied).